### DISCUSSION TITLE
Make error moose installation

### DISCUSSION THREAD

**Discussion by RavisCPP** [idaholab/moose#26995] [Q&A General]
I had to reinstall moose on my system. So, I already have full anaconda setup installed as I am using some other environments using it.
I am using ubuntu 20.04.
So, previously I used to install moose as per the steps provided on this page:

https://mooseframework.inl.gov/getting_started/installation/conda.html

But now even after following the installation process step by step, the set up fails at this step:

(moose) trivedi@trivedi:~/projects/moose/test$ make -j 4

Rebuilding symlinks in /home/trivedi/projects/moose/framework/build/header_symlinks
Creating Unity Directory /home/trivedi/projects/moose/framework/build/unity_src
Copying default MOOSE configuration to: /home/trivedi/projects/moose/framework/include/base/MooseConfig.h...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_stringpiece.cc...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcrecpp.cc...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_get.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_newline.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_string_utils.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_maketables.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_exec.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_ord2utf8.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_refcount.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_tables.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_version.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_fullinfo.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_jit_compile.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_valid_utf8.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_config.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_compile.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_xclass.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_study.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_chartables.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_dfa_exec.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_globals.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_ucd.c...
Compiling C (in opt mode) /home/trivedi/projects/moose/framework/contrib/pcre/src/pcre_byte_order.c...
Compiling C++ /home/trivedi/projects/moose/framework/contrib/gtest/gtest-all.cc...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/hit/parse.cc...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/hit/lex.cc...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/hit/braceexpr.cc...
Building and linking /home/trivedi/projects/moose/framework/contrib/hit/hit.so...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/add_to_log.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/exodiff.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/Tolerance.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/stringx.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/exo_block.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/create_file.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/FileInfo.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/ED_SystemInterface.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/util.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/node_set.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/exoII_read.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/side_set.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/SL_tokenize.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/check.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/iqsort.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/GetLongOpt.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/STRINGLIB_tokenize.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/map.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/contrib/exodiff/exo_entity.C...
Creating Unity Directory /home/trivedi/projects/moose/test/build/unity_src
Rebuilding symlinks in /home/trivedi/projects/moose/test/build/header_symlinks
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/meshgenerators_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/ics_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/markers_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/partitioner_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/kernels_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/fvkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/executors_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/indicators_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/dirackernels_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/nodalkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/scalarkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/reporters_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/interfacekernels_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/samplers_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/functions_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/vectorpostprocessors_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/executioners_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/auxkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/materials_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/actions_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/outputs_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/timesteppers_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/userobjects_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/libtorch_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/fviks_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/controls_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/bcs_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/fvbcs_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/meshes_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/dgkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/constraints_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/postprocessors_Unity.C
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/problems_Unity.C
Creating Resource file /home/trivedi/projects/moose/test/moose_test.yaml
Checking if header needs updating: /home/trivedi/projects/moose/framework/include/base/MooseRevision.h...
make[1]: Entering directory '/home/trivedi/projects/moose/framework/contrib/hit'
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/linesearches_Unity.C
Symlinking MOOSE configure /home/trivedi/projects/moose/framework/build/header_symlinks/MooseConfig.h
Building hit for python with python3-config
mpicxx -std=c++17 -I/home/trivedi/anaconda3/envs/moose/wasp/include -g -fvisibility-inlines-hidden  -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/trivedi/anaconda3/envs/moose/include -std=c++17 main.cc parse.cc lex.cc braceexpr.cc -o hit -Wl,-rpath,/home/trivedi/anaconda3/envs/moose/wasp/lib -L/home/trivedi/anaconda3/envs/moose/wasp/lib -lwaspcore -lwaspddi -lwaspexpr -lwasphalite -lwasphit -lwasphive -lwaspjson -lwasplsp -lwaspsiren -lwaspson
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/positions_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/base_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/interfaces_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/times_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/fvics_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/meshgenerators_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/ics_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/markers_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/partitioner_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/mesh_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/kernels_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/loops_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/fvkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/executors_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/restart_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/indicators_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/timeintegrators_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/physics_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/dirackernels_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/nodalkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/scalarkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/transfers_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/reporters_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/interfacekernels_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/samplers_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/functions_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/vectorpostprocessors_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/executioners_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/auxkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/materials_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/actions_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/outputs_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/functormaterials_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/multiapps_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/timesteppers_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/userobjects_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/dampers_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/limiters_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/preconditioners_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/libtorch_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/splits_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/relationshipmanagers_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/predictors_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/fviks_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/utils_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/bounds_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/controls_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/bcs_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/distributions_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/fvbcs_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/dgkernels_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/systems_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/meshdivisions_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/constraints_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/parser_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/postprocessors_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/geomsearch_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/variables_Unity.C
Creating Unity /home/trivedi/projects/moose/framework/build/unity_src/problems_Unity.C
Linking Library /home/trivedi/projects/moose/framework/contrib/pcre/libpcre-opt.la...
Linking Library /home/trivedi/projects/moose/framework/contrib/gtest/libgtest.la...
Linking Library /home/trivedi/projects/moose/framework/contrib/hit/libhit-opt.la...
Linking Executable /home/trivedi/projects/moose/framework/contrib/exodiff/exodiff...
Creating Unity /home/trivedi/projects/moose/test/build/unity_src/base_Unity.C
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libcrypt.so.1: undefined reference to `getentropy@GLIBC_2.25'
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/build/unity_src/linesearches_Unity.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/framework/build/unity_src/positions_Unity.C...
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2: undefined reference to `fcntl64@GLIBC_2.28'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libp11-kit.so.0: undefined reference to `reallocarray@GLIBC_2.26'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libssh.so.4: undefined reference to `glob@GLIBC_2.27'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libgnutls.so.30: undefined reference to `getrandom@GLIBC_2.25'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libsqlite3.so.0: undefined reference to `log@GLIBC_2.29'
**/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libssh.so.4: undefined reference to `__explicit_bzero_chk@GLIBC_2.25'
collect2: error: ld returned 1 exit status
make: *** [/home/trivedi/projects/moose/framework/moose.mk:449: /home/trivedi/projects/moose/framework/contrib/exodiff/exodiff] Error 1
make: *** Waiting for unfinished jobs....**
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__tdelete@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__twalk_r@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__read_nocancel@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__libc_fcntl64@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__open64_nocancel@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__close_nocancel@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__mprotect@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/librt.so.1: undefined reference to `__recv@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__write_nocancel@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `_IO_enable_locks@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__tfind@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__sigtimedwait@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__tunable_get_val@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__call_tls_dtors@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/librt.so.1: undefined reference to `__libc_pread@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/trivedi/anaconda3/envs/moose/lib/libstdc++.so: undefined reference to `clock_gettime@GLIBC_2.2.5'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__tsearch@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__munmap@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: undefined reference to `__mmap@GLIBC_PRIVATE'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/librt.so.1: undefined reference to `__socket@GLIBC_PRIVATE'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:46: hit] Error 1
make[1]: Leaving directory '/home/trivedi/projects/moose/framework/contrib/hit'
make: *** [/home/trivedi/projects/moose/framework/moose.mk:138: /home/trivedi/projects/moose/framework/contrib/hit/hit.so] Error 2

On trying again, it again fails as:

(moose) trivedi@trivedi:~/projects/moose/test$ make -j 4
Linking Library /home/trivedi/projects/moose/framework/libmoose-opt.la...
Linking Executable /home/trivedi/projects/moose/framework/contrib/exodiff/exodiff...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/test/build/unity_src/base_Unity.C...
Compiling C++ (in opt mode) /home/trivedi/projects/moose/test/build/unity_src/meshgenerators_Unity.C...
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libcrypt.so.1: undefined reference to `getentropy@GLIBC_2.25'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2: undefined reference to `fcntl64@GLIBC_2.28'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libp11-kit.so.0: undefined reference to `reallocarray@GLIBC_2.26'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libssh.so.4: undefined reference to `glob@GLIBC_2.27'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libgnutls.so.30: undefined reference to `getrandom@GLIBC_2.25'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libsqlite3.so.0: undefined reference to `log@GLIBC_2.29'
/home/trivedi/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /usr/lib/x86_64-linux-gnu/libssh.so.4: undefined reference to `__explicit_bzero_chk@GLIBC_2.25'
collect2: error: ld returned 1 exit status
make: *** [/home/trivedi/projects/moose/framework/moose.mk:449: /home/trivedi/projects/moose/framework/contrib/exodiff/exodiff] Error 1
make: *** Waiting for unfinished jobs....

I followed all the steps ranging from installing miniforge as mentioned there even though I had conda installed. I have also tried to directly use complete conda installation directly without miniforge for the purpose, but setup fails both the ways. Can someone help me with the current situation

**Comment by GiudGiud** [comment]
Hello

So this is a re-install in the same folder?
Did you clean the repo (with `git clean -xfD`) before re-installing?

Guillaume

**Reply by RavisCPP** [reply]
yes, I actually cleaned and even removed the folder. The first set of errors come whenever I try fresh install everytime. The second set of errors come when I just try to reinstall on top of first try.
Apologies for confusion. I just wanted to show all the ways I tried to circumvent the problem

**Reply by GiudGiud** [reply]
What is the output of the diagnostics script in moose/scripts

**Reply by milljm** [reply]
Its also possible the issue stems from the  mixing of conda-forge libraries and Anaconda's main channel. To know if this is the case, we would need to see your list of installed Conda packages in the moose environment.

```bash
conda activate moose
conda list
```
What we are looking for, is that _every last package_ is coming from `conda-forge` except of course the moose specific packages, which will be coming from `https://conda.software.inl.gov/public`.

If you see a mix of channels, or blanks. Then that may be the cause. IF that is the case, then:

The fix (according to conda-forge), is to set strict channel priority:
```bash
conda config --set channel_priority strict
```
After which, you should delete the `moose` the environment, and recreate it.
